solr/core/src/java/org/apache/solr/handler/admin/api/CoreReplicationAPI.java

```diff
@@ -46,12 +50,24 @@ public CoreReplicationAPI(SolrCore solrCore, SolrQueryRequest req, SolrQueryResp
 
   @GET
   @Path("/indexversion")
-  @Produces({"application/json", "application/xml", BINARY_CONTENT_TYPE_V2})
+  @Produces({MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML, BINARY_CONTENT_TYPE_V2})
   @PermissionName(CORE_READ_PERM)
   public IndexVersionResponse fetchIndexVersion() throws IOException {
     return doFetchIndexVersion();
   }
 
+  @GET
+  @Path("/files/generation/{gen}")
+  @Produces({MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML, BINARY_CONTENT_TYPE_V2})
+  @PermissionName(CORE_READ_PERM)
+  public NamedList<Object> fetchFiles(
```

Comment:
[0] The method name "fetchFiles" sounds a little bit like you're getting the file-content itself, rather than a list of file names/metadata.  Might be worth updating.

[-0] One of the values of the JAX-RS framework is that we can be explicit about the inputs and outputs of our APIs.  If possible, we should avoid returning Namedlist here and instead return some other strongly typed class that makes it clearer what the response should look like?

Reply:
Changed the method name and returned a POJO object of a strongly typed class. Also updated the tests to reflect that. Let me know what you think.